bugsnag-android-core/src/main/java/com/bugsnag/android/BugsnagVmViolationListener.java

```diff
@@ -0,0 +1,64 @@
+package com.bugsnag.android;
+
+import android.os.Build;
+
+import android.os.StrictMode.OnVmViolationListener;
+import android.os.strictmode.Violation;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+
+import java.util.List;
+
+/**
+ * Sends an error report to Bugsnag for each StrictMode VM policy violation that occurs in
+ * your app.
+ * <p></p>
+ * You should use this class by instantiating Bugsnag in the normal way and then set the
+ * StrictMode policy with
+ * {@link android.os.StrictMode.VmPolicy.Builder#penaltyListener
+ * (Executor, OnVmViolationListener)}.
+ * This functionality is only supported on API 28+.
+ */
+@RequiresApi(api = Build.VERSION_CODES.P)
+public class BugsnagVmViolationListener implements OnVmViolationListener {
+
+    private final Client client;
+    private final OnVmViolationListener listener;
+
+    public BugsnagVmViolationListener(@NonNull Client client) {
+        this(client, null);
+    }
+
+    public BugsnagVmViolationListener(@NonNull Client client,
+                                      @Nullable OnVmViolationListener listener) {
+        this.client = client;
+        this.listener = listener;
+    }
+
+    @SuppressWarnings("ConstantConditions")
+    @Override
+    public void onVmViolation(@NonNull Violation violation) {
+        if (client == null || violation == null) { // bomb out early if supplied client was null
```

Comment:
Same question as above: this effectively swallows events for chained `listener`s